storage/src/test/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/VersionedHashDBSourceTest.java

```diff
@@ -0,0 +1,351 @@
+        .removeVersionedHash(eq(versionedHash.get()));
+  }
+
+  @Test
+  public void computeCanonicalBlobSidecarMetadata_SerializeDeSerialize() {
+    final BlobSidecar blobSidecar = dataStructureUtil.randomBlobSidecar();
+    final SlotAndBlockRootAndBlobIndex expectedBlobIdentifier =
+        blobSidecar.getSlotAndBlockRootAndBlobIndex();
+    final Bytes data = VersionedHashDBSource.computeCanonicalBlobSidecarMetadata(blobSidecar);
+    when(dao.getSidecarIdentifierData(any())).thenReturn(Optional.of(data));
+
+    final Optional<SidecarIdentifier> maybeSidecarIdentifier =
+        versionedHashDBSource.getSidecarIdentifier(versionedHash);
+    verify(dao).getSidecarIdentifierData(any());
+    assertThat(maybeSidecarIdentifier.isPresent()).isTrue();
+    final SidecarIdentifier sidecarIdentifier = maybeSidecarIdentifier.get();
+    assertThat(sidecarIdentifier.canonical()).isTrue();
+    assertThat(sidecarIdentifier.dataColumnSidecarsIdentifierAndBlobIndex()).isEmpty();
+    assertThat(sidecarIdentifier.blobSidecarIdentifier()).contains(expectedBlobIdentifier);
+  }
+
+  @Test
+  public void computeNonCanonicalBlobSidecarMetadata_SerializeDeSerialize() {
+    final BlobSidecar blobSidecar = dataStructureUtil.randomBlobSidecar();
+    final SlotAndBlockRootAndBlobIndex expectedBlobIdentifier =
+        blobSidecar.getSlotAndBlockRootAndBlobIndex();
+    final Bytes data = VersionedHashDBSource.computeNonCanonicalBlobSidecarMetadata(blobSidecar);
+    when(dao.getSidecarIdentifierData(any())).thenReturn(Optional.of(data));
+
+    final Optional<SidecarIdentifier> maybeSidecarIdentifier =
+        versionedHashDBSource.getSidecarIdentifier(versionedHash);
+    verify(dao).getSidecarIdentifierData(any());
+    assertThat(maybeSidecarIdentifier.isPresent()).isTrue();
+    final SidecarIdentifier sidecarIdentifier = maybeSidecarIdentifier.get();
+    assertThat(sidecarIdentifier.canonical()).isFalse();
+    assertThat(sidecarIdentifier.dataColumnSidecarsIdentifierAndBlobIndex()).isEmpty();
+    assertThat(sidecarIdentifier.blobSidecarIdentifier()).contains(expectedBlobIdentifier);
+  }
+
+  @Test
+  public void computeCanonicalDataColumnSidecarMetadata_SerializeDeSerialize() {
+    final DataColumnSidecar dataColumnSidecar = dataStructureUtil.randomDataColumnSidecar();
+    final UInt64 expectedBlobIndex = UInt64.valueOf(123);
+    final Bytes data =
+        VersionedHashDBSource.computeCanonicalDataColumnSidecarMetadata(
+            dataColumnSidecar.getSlotAndBlockRoot(), expectedBlobIndex);
+    when(dao.getSidecarIdentifierData(any())).thenReturn(Optional.of(data));
+
+    final Optional<SidecarIdentifier> maybeSidecarIdentifier =
+        versionedHashDBSource.getSidecarIdentifier(versionedHash);
+    verify(dao).getSidecarIdentifierData(any());
+    assertThat(maybeSidecarIdentifier.isPresent()).isTrue();
+    final SidecarIdentifier sidecarIdentifier = maybeSidecarIdentifier.get();
+    assertThat(sidecarIdentifier.canonical()).isTrue();
```

Comment:
should be able to check the sidecarIdentifier is the same without checking each field?